pkg/controller/nstemplatetier/nstemplatetier_controller.go

```diff
@@ -0,0 +1,222 @@
+	if err != nil {
+		return false, errs.Wrap(err, "unable to get active TemplateUpdateRequests")
+	}
+
+	if activeTemplateUpdateRequests < MaxPoolSize {
+		// create a TemplateUpdateRequest if active count < max pol size
+		// find a MasterUserRecord which is not already up-to-date
+		// and for which there is no TemplateUpdateRequest yet
+		// fetch by subsets of "MaxPoolSize + 1" size until a MasterUserRecord candidate is found
+		murs := toolchainv1alpha1.MasterUserRecordList{}
+		matchingLabels, err := murSelector(tier)
+		if err != nil {
+			return false, errs.Wrap(err, "unable to get MasterUserRecords to update")
+		}
+		err = r.client.List(context.Background(), &murs,
+			client.InNamespace(tier.Namespace),
+			client.Limit(MaxPoolSize+1),
+			matchingLabels,
+		)
+		if err != nil {
+			return false, errs.Wrap(err, "unable to get MasterUserRecords to update")
+		}
+		logger.Info("listed MasterUserRecords", "count", len(murs.Items), "selector", matchingLabels)
+		for _, mur := range murs.Items {
+			// check if there's already a TemplateUpdateRequest for this MasterUserRecord
+			templateUpdateRequest := toolchainv1alpha1.TemplateUpdateRequest{}
+			if err := r.client.Get(context.TODO(), types.NamespacedName{
+				Namespace: tier.Namespace,
+				Name:      mur.Name,
+			}, &templateUpdateRequest); err == nil {
+				logger.Info("MasterUserRecord already has an associated TemplateUpdateRequest", "name", mur.Name)
+				continue
+			} else if !errors.IsNotFound(err) {
+				return false, errs.Wrapf(err, "unable to get TemplateUpdateRequest for MasterUserRecord '%s'", mur.Name)
+			}
+			for _, account := range mur.Spec.UserAccounts {
+				accountTmpls := account.Spec.NSTemplateSet
+				logger.Info("checking MasterUserRecord account", "name", mur.Name, "tier", accountTmpls.TierName)
+				if accountTmpls.TierName != tier.Name {
+					// skip if the user account is not based on the current Tier
+					continue
+				}
+				logger.Info("creating a TemplateUpdateRequest to update the MasterUserRecord", "name", mur.Name, "tier", tier.Name)
+				tur := &toolchainv1alpha1.TemplateUpdateRequest{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: tier.Namespace,
+						Name:      mur.Name,
+						Labels: map[string]string{
+							toolchainv1alpha1.NSTemplateTierNameLabelKey: tier.Name,
+						},
+					},
+					Spec: toolchainv1alpha1.TemplateUpdateRequestSpec{
+						TierName:         tier.Name,
+						Namespaces:       tier.Spec.Namespaces,
+						ClusterResources: tier.Spec.ClusterResources,
+					},
+				}
+				err = controllerutil.SetControllerReference(tier, tur, r.scheme)
+				if err != nil {
+					return false, err
+				}
+				if err = r.client.Create(context.TODO(), tur); err != nil {
+					return false, err
+				}
+				// the controller creates a single TemplateUpdateRequest resource per reconcile loop,
+				// so the request has to be requeued, in order to create other TemplateUpdateRequests
+				// in the next loops
+				return true, nil
```

Comment:
refactored in bd040ce: no explicit `requeue=true` after creating a TemplateUpdateRequest, since the controller already watches such secondary resources.